src/nvapi_gpu.cpp

```diff
@@ -310,11 +386,22 @@ extern "C" {
 
         auto architectureId = adapter->GetArchitectureId();
 
+        if (architectureId > NV_GPU_ARCHITECTURE_GA100) {
+            void* returnAddress = _ReturnAddress();
+
+            // Check if we need to workaround NVIDIA Bug 3634851
+            if (isDLSSBug3634851WARNeeded(returnAddress))
+                architectureId = NV_GPU_ARCHITECTURE_GA100;
+        }
+
         // Assume the implementation ID from the architecture ID. No simple way
         // to do a more fine-grained query at this time. Would need wine-nvml
         // usage.
         NV_GPU_ARCH_IMPLEMENTATION_ID implementationId;
         switch (architectureId) {
+            case NV_GPU_ARCHITECTURE_AD100:
```

Comment:
Cou;d you try to add a unit test for this case to https://github.com/jp7677/dxvk-nvapi/blob/master/tests/nvapi_sysinfo.cpp#L641
(You need to add another test case to `args` and also add  `deviceId` as field to the `args` structure and configure idProps here https://github.com/jp7677/dxvk-nvapi/blob/master/tests/nvapi_sysinfo.cpp#L663)

Reply:
I forgot this earlier, this method also needs to know about the new generation: https://github.com/jp7677/dxvk-nvapi/blob/master/tests/nvapi_system.cpp#L33

Reply:
Here is another place I forgot about earlier https://github.com/jp7677/dxvk-nvapi/blob/master/src/nvapi_d3d12.cpp#L149 
From what I could find, it should be `sm_89` for Ada.